src/components/FullHierarchyTable/FullHierarchyTable.tsx

```diff
@@ -57,6 +76,85 @@ export const FullHierarchyTable: React.FunctionComponent<FullHierarchyTableAllPr
     const fields = props.meta.fields
     const depthLevel = props.depth || 1
     const indentLevel = depthLevel - 1
+    const {t} = useTranslation()
+    const [userOpenedRecords, setUserOpenedRecords] = React.useState([])
+
+    React.useEffect(
+        () => {
+            if (props?.expandedRowKeys) {setUserOpenedRecords(props.expandedRowKeys)}
+        },
+        [props.expandedRowKeys]
+    )
+    const [inputPlaceholder, setInputPlaceholder] = React.useState(props?.searchPlaceholder || '')
+
+    // Apply one filter item
+    const checkFiltered = (filter: BcFilter, checkData: AssociatedItem[]) => {
+        const filteredData: AssociatedItem[] = []
+        checkData?.forEach((dataItem) => {
+            if (dataItem[filter?.fieldName]?.toString().toLowerCase().indexOf(filter?.value?.toString().toLowerCase()) > -1) {
+                if (filter) {
+                    dataItem._searched = true
+                }
+                filteredData.push(dataItem)
+            }
+        })
+        return filteredData
+    }
+
+    // Function check tree to find all parent
+    const findParent = (filteredData: AssociatedItem[], checkData: AssociatedItem[]) => {
+        const parentData: AssociatedItem[] = []
+        filteredData?.forEach((dataItem) => {
+            let currentLevel = dataItem.level as number
+            let currentParentId = dataItem.parentId as string
+            while (currentLevel > 1) {
+                const parentItem = props.data.find((item) => item.id === currentParentId)
+                parentData.push(parentItem)
+                currentParentId = parentItem.parentId as string
+                currentLevel--
+            }
+        })
+        return parentData
+    }
+
+    // Function check tree to find all child
+    const findChild = (filteredData: AssociatedItem[], checkData: AssociatedItem[]) => {
+        // BFS tree find algorithm
```

Comment:
I believe we already have BFS implementation [breadthFirst.ts](https://github.com/tesler-platform/tesler-ui/blob/master/src/utils/breadthFirst.ts), will you able to reused it?

Reply:
The BFS implementation breadthFirst.ts algorithm checks each element for a predicate condition passing deep into the tree by levels.
In the current task, data items provided by a flat list and processing data of all levels at once.
With such an implementation, this is not a difficult operation that can be performed on this component.